Feature for matching assessment

ftbernales opened the pull request:
Added the utils for matching assessment. Originally came from smtk but I did not add the entire library. 

_Main_ script to follow

BTW, what's our plan for file management? My idea is just to put everything under `data/`, but probably would be more organized if there would be a `data/input` and `data/output`?

ftbernales committed:
git ignore

ftbernales committed:
Added sample time-histories

ftbernales committed:
Added sample target spectra

ftbernales committed:
Copied response spectra calculator tool from smtk

ftbernales committed:
Copied strong-motion utilities from smtk

ftbernales committed:
Edit gitignore

ftbernales committed:
Copied IM tools from smtk (commented jit tags)

ftbernales committed:
Added requirements.txt file

stansays commented:
> probably would be more organized if there would be a `data/input` and `data/output`

I agree with this

ftbernales commented:
But how will that impact the dist files? I don't know how to do that yet so no idea on the pre-reqs. I'm only basing on running our scripts on IDE.

stansays commented:
Executable should work the same. All dependencies of all the files are imported, and `./` in the scripts will still correspond to wherever the exe file is placed.

ftbernales committed:
Updated requirements.txt file

ftbernales committed:
Deleted old sample files

ftbernales committed:
Added new sample suite

ftbernales committed:
Added sample target spectra

ftbernales committed:
Initial commit for matching_assessment

ftbernales committed:
Minor edits for dist use

ftbernales commented:
@stansays please help with compiling the code to bundled dist exe, as I haven't done that yet. Are you using _pyinstaller_ for that?

If your time allows, please also help me in fixing bugs here and improving string searches--as commented in some lines. But this will do for our usual naming conventions and practice. I'll just add a README for user guidance.

ftbernales commented:
BTW, can we consider adding config files for user input/interface? This can apply both to Deepsoil-Compile & matching_assessment.

ftbernales committed:
Create README.md

ftbernales commented on src/matching_assessment.py:
```diff
@@ -0,0 +1,480 @@
+    SZ_target = import_SZ_target_spectra(input_dir)
+
+    # Get RotD1000-Component Response Spectra for ASC Suite
+    suite_rotd100_spectra = compute_suite_rotd100_spectra(input_dir, periods,
```
@stansays Will the mp pool be placed here or in the function definition?

stansays committed:
Make case-insensitive and require specified target

The searching for target spectra files was made case-insensitive, and an error is raised if neither ASC nor SZ is specified.

stansays committed:
Added multiprocessing for RotD100 computation

stansays commented:
> @stansays please help with compiling the code to bundled dist exe, as I haven't done that yet. Are you using _pyinstaller_ for that?
> 
> If your time allows, please also help me in fixing bugs here and improving string searches--as commented in some lines. But this will do for our usual naming conventions and practice. I'll just add a README for user guidance.

I use the command
`pyinstaller -F [script]`

Can the `PeerMSR` class and `hazardlib.geo` module from openquake be removed in `sm_utils.py`, or do we intend to use them? I'm hesitant to compile as of now because of the possibility of ballooning the distribution with dependencies from openquake.

stansays committed:
Dictionary union for earlier Python versions

| as dictionary union is only available for Python 3.9+
Changed it to support 3.6+

ftbernales committed:
Updated requirements.txt

ftbernales committed:
Edited input directory location

ftbernales committed:
Updated gitignore

ftbernales committed:
Moved input ATH files from pr-1/stansays/rotd100

ftbernales committed:
Moved input profiles from pr-1/stansays/rotd100

ftbernales committed:
Moved output files from pr-1/stansays/rotd100

ftbernales committed:
Merged Deepsoil-Compile from pr-1/stansays/rotd100

ftbernales committed:
Merged changes from pr-1/stansays/rotd100

ftbernales committed:
Updated requirements.txt

ftbernales committed:
Removed numba import statement

ftbernales commented:
Done merging the files to [matching_assessment](https://github.com/ftbernales/Deepsoil-Compile/tree/matching_assessment)

ftbernales committed:
Edited input dir for dist

ftbernales committed:
Regenerated requirements.txt

stansays committed:
Updated data input and output directories

ftbernales committed:
Updated gitignore

Commit:
Edited output dir for dist

Commit:
Updated gitignore for build and spec files

Commit:
Added print logs and minor edits

Commit:
Minor edit on time logger

Commit:
Deleted buggy function for Windows

Commit:
Fixed for char limit via PEP8

ftbernales committed:
Generated one-file bundled exe

ftbernales committed:
Update README.md

Included sample directory tree for running exe + edited instructions

ftbernales commented:
@stansays The bundled code is now working, but can't get it to work in my AMH PC due to admin restriction + reputation-based protection. 

Kindly review one more time. I think we can merge this if you're good with this (_for now_).

stansays committed:
Update dist file for Deepsoil_Compile
